exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillCursor.java

```diff
@@ -239,6 +261,11 @@ QueryDataBatch getNext() throws UserException, InterruptedException {
             }
             return qdb;
           }
+
+          // Check and throw SQLTimeoutException
+          if ( parent.timeoutInMilliseconds > 0 && parent.elapsedTimer.elapsed(TimeUnit.SECONDS) >= parent.timeoutInMilliseconds ) {
```

Comment:
wrong unit for the comparison (should be millis)

Reply:
Darn! +1